providers/gavc_download.rb

```diff
@@ -14,6 +17,9 @@ def whyrun_supported?
                                                  packaging: new_resource.packaging)
 
   resource_name = "Downloading file #{download_uri} specified by: #{new_resource.to_s}"
+  if new_resource.username && new_resource.password then
+    download_uri.sub!(/^(.*):\/\//, "\\1://#{new_resource.username}:#{new_resource.password}@")
```

Comment:
Sorry, I meant add a unit test to verify the operation of the new username and password attributes. I can try adding one if you aren't familiar with chefspec.